### PR TITLE
Update README with more detailed steps + minor changes

### PR DESCRIPTION
**Pull request by JayOrten**
- Adds detailed steps on data preprocessing in the README
- Adds an additional parameter, `hf_filesystem_path`, for downloading data from the HF filesystem. Before, we were constructing this path, but that's not a good way to do it because it depends on the repo.
- Removes quotation marks from the YAML. See: https://stackoverflow.com/questions/19109912/yaml-do-i-need-quotes-for-strings-in-yaml
- Updates some defaults in the slurm scripts
- Add a progress bar when splitting data

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,6 +70,42 @@ cd scripts/user_scripts
 ./train_model.sh
 ```
 
+The process for running the scripts via Slurm is similar. The corresponding Slurm scripts are located in the [slurm](./slurm/) folder.
+
+More details for these steps are described in the following sections.
```
Comment (KimballNJardine) [inline README.md]: This reads like the following commands relate to Slurm specifically.

File: src/split_data.py [hunk 2]
```diff
@@ -1,4 +1,9 @@
 import datasets
+import dask
+dask.config.set({'dataframe.query-planning': True})
+import dask.dataframe as dd
+from dask.diagnostics import ProgressBar
```
Comment (KimballNJardine) [inline src/split_data.py]: Move this to the from section of import statements.

**Review by KimballNJardine — CHANGES_REQUESTED** [review]
Please make Readme.md clearer that lines below Slurm comment apply to everything, and not just slurm, and move import statement in split_data.py. Otherwise good.

**Review by KimballNJardine — APPROVED** [review]
Looks good.
Approved.

File: README.md [hunk 3]
```diff
@@ -70,6 +72,40 @@ cd scripts/user_scripts
...
+
+##### Downloading Data
+
+This repository is designed to utilize datasets available through the HuggingFace Hub. There are two ways to download data:
+
+1. [download_data.py](./src/download_data.py), run via the [download_data.sh](./scripts/download_data.sh) script, can download data through the HuggingFace Filesystem. To do this, ensure that the correct parameters are set in the configuration YAML, specifically `hf_filesystem_path`. This must correspond to the correct HF Filesystem path. This method is good for smaller datasets that can easily fit in memory.
+
+2. By cloning the HuggingFace Dataset repo directly, and downloading the necessary data. [download_c4.sh](./scripts/download_c4.sh) exists as an example of this for the C4 dataset. This method is good for very large datasets.
+
+It should be noted that datasets can come in a variety of different formats. Currently, this repo works best with Parquet files. If the data is downloaded in a different format, the code may need to be changed to accomodate.
```
Comment (tristancorncroissant) [inline README.md]: Link the Parquet files if we're going to capitalize it IMO https://parquet.apache.org/

File: README.md [hunk 4]
```diff
@@ -70,6 +72,40 @@ cd scripts/user_scripts
...
+
+#### Data Preprocessing
+
+This repository uses [Dask](https://docs.dask.org/en/stable/) to load and process data. Our code is configured to load datasets into Dask from Parquet files. If using a different file format, this may need to be changed.
+
+##### Downloading Data
+
+This repository is designed to utilize datasets available through the HuggingFace Hub. There are two ways to download data:
+
+1. [download_data.py](./src/download_data.py), run via the [download_data.sh](./scripts/download_data.sh) script, can download data through the HuggingFace Filesystem. To do this, ensure that the correct parameters are set in the configuration YAML, specifically `hf_filesystem_path`. This must correspond to the correct HF Filesystem path. This method is good for smaller datasets that can easily fit in memory.
```
Comment (tristancorncroissant) [inline README.md]: Start with "By running [download_data.py](./src/download_data.py) via the..."

File: README.md [hunk 5]
```diff
@@ -70,6 +72,40 @@ cd scripts/user_scripts
...
+
+2. By cloning the HuggingFace Dataset repo directly, and downloading the necessary data. [download_c4.sh](./scripts/download_c4.sh) exists as an example of this for the C4 dataset. This method is good for very large datasets.
+
+It should be noted that datasets can come in a variety of different formats. Currently, this repo works best with Parquet files. If the data is downloaded in a different format, the code may need to be changed to accomodate.
+
+##### Splitting the Data
+
+After downloading the data, you can split the data into separate train/validation/test splits via the [split_data.py](./src/split_data.py) script, run through [split_data.sh](./scripts/split_data.sh).
+
+Optionally, within [split_data.py](./src/split_data.py), you can specify to shuffle the data while splitting. This is more expensive for larger datasets. Any other preprocessing and pretokenization steps should occur within this step.
```
Comment (tristancorncroissant) [inline README.md]: Replace 

"Any other preprocessing and pretokenization steps should occur within this step." 

with

"This and any other preprocessing or pretokenization steps should occur here prior to splitting the data."

File: README.md (hunk 4 above)
Comment (tristancorncroissant) [inline README.md]: Replace 

"download data through the HuggingFace Filesystem."

with

"download data through the [HuggingFace Filesystem].(https://huggingface.co/docs/huggingface_hub/v0.22.0.rc0/en/package_reference/hf_file_system#huggingface_hub.HfFileSystem)"

**Review by DrewGalbraith — CHANGES_REQUESTED** [review]

**Review by DrewGalbraith — APPROVED** [review]

**Pull request merged**